src/components/market/AllProducts.js

```diff
@@ -0,0 +1,76 @@
+import React, { useEffect, useState } from "react";
+import styles from "styles/markets.module.scss";
+import useDeviceState from "features/hooks/useDeviceState";
+import Card from "./Card";
+import Button from "components/commons/Button";
+import { SearchInput, SelectInput } from "components/commons/Inputs";
+import { useRecoilValue, useSetRecoilState } from "recoil";
+import { currentPageState, totalPagesState } from "context/atoms/page";
+import { itemsOrderState } from "context/atoms/order";
+import getProductsPerPage from "features/utils/getProductsPerPage";
+import getProductsData from "apis/getProductsData";
+import useResetPage from "features/hooks/useResetPage";
+
+export default function AllProducts() {
+  const [keyword, setKeyword] = useState("");
+  const [totalCount, setTotalCount] = useState(0);
+  const [renderDataList, setRenderedDataList] = useState([]);
+  const orderState = useRecoilValue(itemsOrderState);
+  const currentPage = useRecoilValue(currentPageState);
+  const setTotalPages = useSetRecoilState(totalPagesState);
+  const deviceState = useDeviceState();
+  useResetPage([deviceState, keyword]);
+
+  // 렌더되는 데이터 설정
+  useEffect(() => {
+    (async () => {
+      const productsPerPage = getProductsPerPage(deviceState);
+      const order = orderState === "최신순" ? "recent" : "favorite";
+
+      const data = await getProductsData({
+        order,
+        page: currentPage,
+        pageSize: productsPerPage,
+        keyword,
+      });

```

Comment:
이 useEffect에서도 currentPage, keyword, pageSize가 없는 경우에는 객체에 내용이 비어있는 프로퍼티를 넘겨주지않게끔 해주면 좋겠죠? :) 간단하게 spread operator를 통해 처리해줄수있답니다!

```suggestion
      const defaultValues = { order, keyword }; 
      const data = await getProductsData({
        ...defaultValues,
        ...(currentPage && { page: currentPage }),
        ...(productsPerPage && { pageSize: productsPerPage }),
      });

```

Reply:
넵 확인했습니다!